map_gui/src/tools/city_picker.rs

```diff
@@ -504,11 +472,27 @@ impl<A: AppLike + 'static> State<A> for CitiesInCountryPicker<A> {
 
 fn cities_per_country() -> BTreeMap<String, Vec<CityName>> {
     let mut per_country = BTreeMap::new();
-    for city in CityName::list_all_cities_from_system_data() {
+    for city in CityName::list_all_cities_from_manifest(&Manifest::load()) {
         per_country
             .entry(city.country.clone())
             .or_insert_with(Vec::new)
             .push(city);
     }
     per_country
 }
+
+fn chose_city<A: AppLike + 'static>(
```

Comment:
Both the code and UX of this aren't great. UX-wise, you're sometimes prompted to download missing stuff. If you agree and it succeeds, the popup message tells you to just repeat whatever you just tried to open -- it doesn't automatically start the challenge, or the santa level, or whatever else. And then in the code perspective, all these places loading a new map have to first check for the local file and maybe download.

What I'd like to do instead is automatically load the map after downloading it and not make the player click something again. Coding it is quite complex, though -- `MapLoader` takes this callback to execute after a map is loaded successfully. That's the `on_load` thing we plumb around here. Making that work with `prompt_to_download_missing_data` is hard because of lifetimes / closures, and because there are some failure modes (can't download the map, player chooses to not download). I attempted to work through this, but got stuck.

One idea I'm toying with is to remove the `on_load` callback from `MapLoader` entirely. When it's done, it'll just `Transition::Pop` back to the original state that asked it to load something. Then somehow, we can check the success/failure of the load and do the appropriate state transition. I'd like to just write the "synchronous" style code:

```
match MapLoader::load(ctx, app, map_name) {
  Ok(()) => { // Now the new map is loaded, go do whatever }
  Err(err) => { // Still on the same map we were originally }
  }
```

I'll play around with some ideas for doing this.